boofuzz/sessions.py

```diff
@@ -140,10 +140,10 @@ def send(self, data):
         Returns:
             None
         """
-        if self._fuzz_data_logger is not None:
-            self._fuzz_data_logger.log_send(data)
```

Comment:
I'm leaning toward keeping some kind of log message prior to the send action. It helps in the case where something fails while the send is happening.

Likewise, if we're only logging what is actually sent, we can get rid of the log action `self._fuzz_data_logger.log_info("{0} bytes sent".format(num_sent))` just a few lines below, as that information is now redundant. In fact, we could just move those two lines right here and change the string to "Sending {0} bytes".

Reply:
I made some changes according to your ideas, please check.